pkg/vm/engine/tae/moengine/reader.go

```diff
@@ -58,7 +58,10 @@ func (r *txnReader) Read(refCount []uint64, attrs []string) (*batch.Batch, error
 	if n > cap(r.zs) {
 		r.zs = make([]int64, n)
 	}
-	bat.Zs = r.zs[:n]
+	bat.Zs = make([]int64, n)
+	for i := 0; i < n; i++ {
+		bat.Zs[i] = r.zs[i]
```

Comment:
Why use new []int64 instead of range of origin slice

Reply:
to solve the data race, two thread use the same Zs